trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/credentials/http/TestHttpCredentialsProvider.java

```diff
@@ -123,6 +134,14 @@ public void testIncorrectResponseFromServer()
         assertThat(actual).isEmpty();
     }
 
+    @Test void testValidCredentialsIncorrectHeader()
```

Comment:
This test is requiring us to maintain an http client, and this test case is pretty much covered by the fact all other tests check that the request contains the right headers.

We also have tests to check the config is parsed properly.

I think it'd be fine to remove this test in the interest of cleanliness